### PR TITLE
STAC 13243: split error into 4xx and 5xx for xray

### PR DESCRIPTION
**Pull request by vishalxebia**
### Step 3: Did you add / update tests for your changes in the right area?
- [ ] Unit/Component test
- [ ] Integration test	

### Step 4: I'm confident that everything is properly tested:
I got a PO / QA Approval by:
- [ ] Name

### Step 5: Can we ship this feature to production?
- [ ] Yes, I'm proud of my work. Ship it! :ship:

### CODE REVIEW

File: aws_xray/stackstate_checks/aws_xray/aws_xray.py [hunk 1]
```diff
@@ -343,9 +342,31 @@ def flatten_segment(segment, kind):
...
+    Process the span on different error code
+    :param span: The span in which error code has to be sent
+    :param segment: The segment from which error code has to be processed
+    :return: span processed with error codes
+    """
+    error = segment.get('fault') or segment.get('throttle') or segment.get('error')
+    if is_affirmative(error):
+        flat_meta = span.get('meta')
+        if flat_meta:
+            error_code = flat_meta.get("http.response.status")
```
Comment (TizianoPerrucci) [inline aws_xray/stackstate_checks/aws_xray/aws_xray.py:360]: this is a `status_code`

File: aws_xray/stackstate_checks/aws_xray/aws_xray.py [hunk 2]
```diff
@@ -343,9 +342,31 @@ def flatten_segment(segment, kind):
...
+def process_http_error(span, segment):
+    """
+    Process the span on different error code
+    :param span: The span in which error code has to be sent
+    :param segment: The segment from which error code has to be processed
+    :return: span processed with error codes
+    """
+    error = segment.get('fault') or segment.get('throttle') or segment.get('error')
+    if is_affirmative(error):
+        flat_meta = span.get('meta')
```
Comment (TizianoPerrucci) [inline aws_xray/stackstate_checks/aws_xray/aws_xray.py:358]: why the variable is called `flat_meta` ?

File: aws_xray/stackstate_checks/aws_xray/aws_xray.py [hunk 3]
```diff
@@ -343,9 +342,31 @@ def flatten_segment(segment, kind):
...
 
 
+def process_http_error(span, segment):
+    """
+    Process the span on different error code
+    :param span: The span in which error code has to be sent
+    :param segment: The segment from which error code has to be processed
+    :return: span processed with error codes
+    """
+    error = segment.get('fault') or segment.get('throttle') or segment.get('error')
```
Comment (TizianoPerrucci) [inline aws_xray/stackstate_checks/aws_xray/aws_xray.py:356]: what is the meaning of `throttle` ?

**Review by TizianoPerrucci — CHANGES_REQUESTED** [review]
I also miss tests for segments containing `fault` or `throttle`

File: aws_xray/stackstate_checks/aws_xray/aws_xray.py (hunk 3 above)
Comment (vishalxebia) [reply aws_xray/stackstate_checks/aws_xray/aws_xray.py:356]: @TizianoPerrucci  this is the `errortype` returned by `AWS` for status code 429.

**Review by TizianoPerrucci — APPROVED** [review]

**Pull request merged**